src/mlpack/methods/ann/layer/not_adapted/parametric_relu_impl.hpp

```diff
@@ -21,66 +21,66 @@
 namespace mlpack {
 namespace ann /** Artificial Neural Network. */ {
 
-template<typename InputDataType, typename OutputDataType>
-PReLU<InputDataType, OutputDataType>::PReLU(
+template<typename InputType, typename OutputType>
+PReLUType<InputType, OutputType>::PReLUType(
     const double userAlpha) : userAlpha(userAlpha)
 {
   alpha.set_size(WeightSize(), 1);
   alpha(0) = userAlpha;
 }
 
-template<typename InputDataType, typename OutputDataType>
-void PReLU<InputDataType, OutputDataType>::Reset()
+template<typename InputType, typename OutputType>
+void PReLUType<InputType, OutputType>::SetWeights(
+    typename OutputType::elem_type* weightsPtr)
 {
+  alpha = arma::mat(weightsPtr, 1, 1, false, false);
+
   //! Set value of alpha to the one given by user.
-  alpha = arma::mat(alpha.memptr(), 1, 1, false, false);
+  // TODO: this doesn't even make any sense.  is it trainable or not?
+  // why is there userAlpha?  is that for initialization only?
+  alpha(0) = userAlpha;
 }
 
-template<typename InputDataType, typename OutputDataType>
 template<typename InputType, typename OutputType>
-void PReLU<InputDataType, OutputDataType>::Forward(
+void PReLUType<InputType, OutputType>::Forward(
     const InputType& input, OutputType& output)
 {
-  output = arma::max(input, alpha(0) * input);
+  // TODO: use transform()?
+  output = input;
+  arma::uvec negative = arma::find(input < 0);
+  output(negative) = input(negative) * alpha(0);
```

Comment:
This seems to be overriden from #3115

Reply:
Do you mean that I haven't merged in changes from #3115 properly?  That's probably true---this branch has become out of sync.  If you want to do it (for this issue and in the few other places where it happens), feel free to open a PR to this branch with the changes.  Otherwise, I'll go through and (try to) sync everything up once this is closer to ready.  (I think it's getting close!)